### PR TITLE
fix(raiko): unsafe align vec to avoid unalign mem access

### PR DESCRIPTION
**Pull request by smtmfft**

### CODE REVIEW

File: lib/src/protocol_instance.rs [hunk 1]
```diff
@@ -40,20 +41,36 @@ impl ProtocolInstance {
                 input.taiko.tx_blob_hash.unwrap()
             } else {
                 println!("kzg check enabled!");
-                let mut data = Vec::from(KZG_TRUST_SETUP_DATA);
-                let kzg_settings = KzgSettings::from_u8_slice(&mut data);
+                let data_size = KZG_TRUST_SETUP_DATA.len();
+                let aligned_data_size = data_size + (4 - (data_size % 4));
```
Comment (Brechtpd) [inline lib/src/protocol_instance.rs:45]:
```suggestion
                let aligned_data_size = (data_size + 3) / 4 * 4;
```

Doesn't really matter, but this method seems more standard and also doesn't increase the size needlessly by 4 when `data_size % 4 == 0`.

File: lib/src/protocol_instance.rs [hunk 2]
```diff
@@ -40,20 +41,36 @@ impl ProtocolInstance {
...
+
+                let kzg_settings = KzgSettings::from_u8_slice(&mut aligned_vec);
                 let kzg_commit = KzgCommitment::blob_to_kzg_commitment(
                     &Blob::from_bytes(input.taiko.tx_data.as_slice())
                         .expect("Fail to form blob from tx bytes"),
                     &kzg_settings,
                 )
                 .expect("Fail to calculate KZG commitment");
                 let versioned_hash = kzg_to_versioned_hash(&kzg_commit);
+                println!("kzg check !versioned_hash = {:?}", versioned_hash);
```
Comment (Brechtpd) [inline lib/src/protocol_instance.rs:67]: remove?

**Review by Brechtpd — APPROVED** [review]
I haven't tested it yet, but if this fixes the problem than that actually makes a lot of sense!

File: lib/src/protocol_instance.rs (hunk 2 above)
Comment (smtmfft) [reply lib/src/protocol_instance.rs:67]: will do, just a POC fix now.

**Pull request merged**